lib/node_modules/@stdlib/math/base/special/gammasgn/lib/main.js

```diff
@@ -0,0 +1,70 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var floor = require( '@stdlib/math/base/special/floor' );
+
+
+// MAIN //
+
+/**
+* Computes the principal square root of the product of 𝛑 and a positive number.
+*
+* @param {number} x - input value
+* @returns {number} results
+*
+* @example
+* var v = gammasgn( 1.0 );
+* // returns 1.0
+*
+* v = gammasgn( -2.5 );
+* // returns -1.0
+*
+* v = gammasgn( 0.0 );
+* // returns 0.0
+*
+* v = gammasgn( NaN );
+* // returns NaN
+*/
+function gammasgn( x ) {
+	var z;
+
+	if ( isnan( x ) ) {
+		return x;
+	}
+	if ( x > 0 ) {
+		return 1.0;
+	}
+	z = floor( x );
+	if ( x - z === 0 ) {
+		return 0.0;
+	}
+	if ( parseInt( z, 10 ) % 2 ) {
+		return -1.0;
+	}
+	return 1.0;
```

Comment:
```suggestion
	var fx;
	if ( isnan( x ) ) {
		return x;
	}
	if ( x > 0 ) {
		return 1.0;
	}
	fx = floor( x );
	if ( x === fx ) {
		return 0.0;
	}
	fx /= 2.0;
	if ( fx === floor( fx ) ) {
		return -1.0;
	}
	return 1.0;
```

The `x - fx === 0` check can be simplified to an equality check. An optimizing compiler could simplify and rearrange on its own, but I'd rather not leave it to chance.

`parseInt` is not necessary here. True, a native implementation might cast to int and then use the modulus. However, the modulus can be a slower operation. The basic goal of the line in question is checking whether the value is even. We can do that with a division (still slower) and a `floor`. Still a bit slow, but follows https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/assert/is-even/lib/is_even.js.

Reply:
In fact it is reviewing that the floor of the given number is odd, I added a call to the function in stdlib, let me know if we don't want that dependency so I can change it again

Reply:
No need to use the dependency. Just do `fx !== floor( fx )`.

Reply:
Thanks for catching that. You're right regarding the "is odd" check.

Although, we should just do this instead:

```
if ( fx === floor( fx ) ) {
    return 1.0;
}
return -1.0;
```

meaning just swap the return values.